### PR TITLE
Add material json description serialisation

### PR DESCRIPTION
**Pull request by rchristie**
Closes issue #176.
Closes issue #154.

### CODE REVIEW

File: src/graphics/material.cpp [hunk 1]
```diff
@@ -534,175 +530,154 @@ static int Graphical_material_remove_module_if_matching(cmzn_material *material,
...
+		double alpha;
+		double shininess;
 	}
+	default_material = {
+		/*ambient*/ { 1.00, 1.00, 1.00},
+		/*diffuse*/ { 1.00, 1.00, 1.00},
+		/*emission*/{ 0.00, 0.00, 0.00},
+		/*specular*/{ 0.00, 0.00, 0.00},
+		/*alpha*/1.0,
+		/*shininess*/0.0 },
```
Comment (hsorby) [inline src/graphics/material.cpp]: Should this be a semicolon here?

**Review by hsorby — CHANGES_REQUESTED** [review]
A minor change, and a query about he default surface material should it be defined like default material and default selected material are?  Is it set in the initialisation of the material module?

File: src/graphics/material.cpp (hunk 1 above)
Comment (rchristie) [reply src/graphics/material.cpp]: It is legal, but I don't like multiple assignments when declaring variables (like int a=1, b=2;) , especially not with structures in this way. I'll tidy it up.

**Comment by rchristie** [comment]
> A minor change, and a query about he default surface material should it be defined like default material and default selected material are? Is it set in the initialisation of the material module?

The default surface material is initially null which gives the older behaviour where the default material is used. I added it so that when we make the default material black (good for lines on a white background), a different default may be used for surfaces and contours. Note I'm referring to the default material attributes of the Material module here, not the material called 'default'.
If we can come up with a better plan for these it can be added in this PR.

File: src/description_io/material_json_io.hpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+ *
+ */
+/* OpenCMISS-Zinc Library
+*
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#if !defined (TESSELLATION_JSON_IO_HPP)
+#define TESSELLATION_JSON_IO_HPP
```
Comment (alan-wu) [inline src/description_io/material_json_io.hpp]: Surely this is not meant to be tessellation.

Comment (alan-wu) [reply src/description_io/material_json_io.hpp]: I would be surprised if both exports work at the same time.

**Review by alan-wu — CHANGES_REQUESTED** [review]

File: src/description_io/material_json_io.hpp (hunk 2 above)
Comment (rchristie) [reply src/description_io/material_json_io.hpp]: Fixed.

File: src/graphics/material.cpp (hunk 1 above)
Comment (rchristie) [reply src/graphics/material.cpp]: Tidied up code for defining default and standard materials.

**Review by hsorby — APPROVED** [review]

**Review by alan-wu — APPROVED** [review]

**Pull request merged**